### PR TITLE
chore: use embedded logger for Ingester & Flusher structs

### PR DESCRIPTION
**Pull request by jtlisi**
**What this PR does**:

- Adds a logger field for both the ingester and flusher structs

**Which issue(s) this PR fixes**:

A TODO already exists in the code base and I created an issue #3778 

**Checklist**
- [x] Tests updated

### CODE REVIEW

**Review by pracucci — COMMENTED** [review]
Thanks! A couple of things:
1. May you fix the building issues?
2. What do you think about introducing a `faillint` rule (in the `Makefile`) to forbid the usage of `util.Logger` and `util_log.Logger` from the packages from which you removed it?

**Review by pracucci — COMMENTED** [review]
I've just merged https://github.com/cortexproject/cortex/pull/3781. Could you rebase, please? Thanks a lot!

**Comment by jtlisi** [comment]
> May you fix the building issues?
👍 

> What do you think about introducing a faillint rule (in the Makefile) to forbid the usage of util.Logger and util_log.Logger from the packages from which you removed it?

Added. I really like this approach and it should make it easy to piecemeal remove the global logger from packages.

File: Makefile [hunk 1]
```diff
@@ -180,6 +180,9 @@ lint:
 		./pkg/frontend/... \
 		./pkg/querier/tenantfederation/... \
 		./pkg/querier/queryrange/...
+	# Ensure packages that no longer use a global logger don't reintroduce it
+	faillint -paths "github.com/cortexproject/cortex/pkg/util/log.{Logger}" \
+		./pkg/ingester/...
```
Comment (pracucci) [inline Makefile:185]: And `./pkg/flusher/...` too.

File: pkg/ingester/mapper.go [hunk 2]
```diff
@@ -105,8 +107,8 @@ func (m *fpMapper) maybeAddMapping(
 		// A new mapping has to be created.
 		mappedFP = m.nextMappedFP()
 		mappedFPs[ms] = mappedFP
-		level.Debug(util_log.Logger).Log(
-			"msg", "fingerprint collision detected, mapping to new fingerprint",
+		level.Debug(m.logger).Log(
+			"msg", "ingerprint collision detected, mapping to new fingerprint",
```
Comment (pracucci) [inline pkg/ingester/mapper.go:111]: Whoops

```suggestion
			"msg", "fingerprint collision detected, mapping to new fingerprint",
```

File: pkg/ingester/user_state.go [hunk 3]
```diff
@@ -128,6 +132,7 @@ func (us *userStates) getOrCreate(userID string) *userState {
 	state, ok := us.get(userID)
 	if !ok {
 
+		logger := log.With(us.logger, "user", userID)
```
Comment (pracucci) [inline pkg/ingester/user_state.go:135]: 👍

**Review by pracucci — APPROVED** [review]
Good job, LGTM (modulo a couple of nits)

**Pull request merged**